src/python/pants/backend/python/subsystems/debugpy.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.backend.python.goals import lockfile
+from pants.backend.python.goals.lockfile import GeneratePythonLockfile
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import EntryPoint
+from pants.core.goals.generate_lockfiles import GenerateToolLockfileSentinel
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import IntOption, StrOption
+from pants.util.docutil import git_url
+
+
+class DebugPy(PythonToolBase):
+    options_scope = "debugpy"
+    help = "An implementation of the Debug Adapter Protocol for Python (https://github.com/microsoft/debugpy)."
+
+    default_version = "debugpy==1.6.0"
+    default_main = EntryPoint("debugpy")
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7,<3.11"]
```

Comment:
This is slightly odd, because this needs to safely mix with the interpreter for `pytest` and your code. Ideally this would inherit the exact same constraints as `pytest` (i.e. they need to be in the "same resolve"), but I can't picture how to do that via our existing subsystem infrastructure.

Reply:
At a minimum, should this be set to `Pytest.default_interpreter_constraints`?

Reply:
Well this isn't test specific: https://github.com/pantsbuild/pants/pull/15829